source/components/JsxParser.js

```diff
@@ -120,6 +120,15 @@ export default class JsxParser extends Component {
           case '!': return !expression.argument.value
         }
         return undefined
+      case 'ArrowFunctionExpression':
+        if (expression.async || expression.generator) {
+          this.props.onError(new Error('Async and generator arrow functions are not supported.'))
```

Comment:
Should probably return `undefined` here, as well - to short-circuit further processing of this expression.